### DISCUSSION TITLE
Loading adafruit libraries/drivers into MP project?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16374] [General]
Loading adafruit libraries/drivers into MP project? Via Thonny if possible?  Searching for adafruit nothing is returned?

**Comment by andrewleech** [comment]
Are you referring to circuitpython libraries?
Unfortunately there's enough differences between circuitpython and micropython that these libraries usually don't work without some adapting.

**Comment by scruss** [comment]
[Adafruit_Blinka](https://github.com/adafruit/Adafruit_Blinka) is supposed to be an interface shim between MicroPython and CircuitPython libraries. Never tried it myself. You may need [Adafruit_Python_PlatformDetect](https://github.com/adafruit/Adafruit_Python_PlatformDetect) too. Installation guide, supposedly: [Installing Blinka and Libraries | CircuitPython Libraries on MicroPython using the Raspberry Pi Pico | Adafruit Learning System](https://learn.adafruit.com/circuitpython-libraries-on-micropython-using-the-raspberry-pi-pico/installing-blinka-and-libraries)

If this is to support particular Adafruit hardware, there's a small number of pre-schism Adafruit `micropython_*` libraries that haven't been ported. The code for these likely makes some rather out-of-date assumptions, so may not be too much fun to get working with current versions of MicroPython

**Comment by Jibun-no-Kage** [comment]
Well that is pickle... in that I wanted to support the TCS34725 sensor as I already support the VEML6040 on some of my ESP32 and ESP8266 projects.

Specific to TCS34725 sensor, I found 2 versions of the adafruit code.  Version 1 depreciated, but I rewrote it a bit and it works, at least as well as I need it on ESP32 and ESP822 testing thus far.  The newer version, i..e 2, maybe better, but requires or needs extensive code based on the adafruit common libraries.  So I am at least trying to understand how to support the adafruit libraries on MP.  Per adafruit it is possible.

Based on what you are suggesting, it might be easier to just run MP and rewrite stuff as needed, even if based on adafruit sources.  Or run Circuit Python out right.

**Reply by mattytrentini** [reply]
The [Adafruit_CircuitPython_TCS34725](https://github.com/adafruit/Adafruit_CircuitPython_TCS34725) looks in reasonable shape - and seems to be trivially easy to port to MicroPython.

**Reply by scruss** [reply]
[Awesome MicroPython](https://awesome-micropython.com/#colour) lists Rob Hamerling's [micropython-tcs34725](https://gitlab.com/robhamerling/micropython-tcs34725) driver. It's limited, but may do what you want.

Awesome MicroPython is frequently Awesome.
